test/lib/getWorkflow.test.js

```diff
@@ -21,72 +21,179 @@ describe('getWorkflow', () => {
+            pipelineConfig: REQUIRES_WORKFLOW,
+            triggerFactory: triggerFactoryMock
+        });
+        const { pipelineWorkflow: legacyRequires } = await getWorkflow({
+            pipelineConfig: LEGACY_AND_REQUIRES_WORKFLOW,
+            triggerFactory: triggerFactoryMock
+        });
+        const { pipelineWorkflow: external } = await getWorkflow({
+            pipelineConfig: EXTERNAL_TRIGGER,
+            triggerFactory: triggerFactoryMock
+        });
 
         assert.deepEqual(requires, EXPECTED_OUTPUT);
         assert.deepEqual(legacyRequires, EXPECTED_OUTPUT);
         assert.deepEqual(external, EXPECTED_EXTERNAL);
     });
 
     it('should handle displayName', async () => {
-        const result = await getWorkflow(
-            {
-                jobs: {
-                    foo: {
-                        annotations: {
-                            'screwdriver.cd/displayName': 'baz'
-                        }
-                    },
-                    bar: {}
-                }
-            },
-            triggerFactoryMock
-        );
+        const pipelineConfig = {
+            jobs: {
+                foo: {
+                    annotations: {
+                        'screwdriver.cd/displayName': 'baz'
+                    }
+                },
+                bar: {}
+            }
+        };
+        const { pipelineWorkflow: result } = await getWorkflow({
+            pipelineConfig,
+            triggerFactory: triggerFactoryMock
+        });
 
         assert.deepEqual(result, {
             nodes: [{ name: '~pr' }, { name: '~commit' }, { name: 'foo', displayName: 'baz' }, { name: 'bar' }],
             edges: []
         });
     });
 
-    it('should handle detatched jobs', async () => {
-        const result = await getWorkflow(
-            {
-                jobs: {
-                    foo: {},
-                    bar: { requires: ['foo'] }
-                }
-            },
-            triggerFactoryMock
-        );
+    it('should handle detached jobs', async () => {
+        const pipelineConfig = {
+            jobs: {
+                foo: {},
+                bar: { requires: ['foo'] }
+            }
+        };
+        const { pipelineWorkflow: result } = await getWorkflow({
+            pipelineConfig,
+            triggerFactory: triggerFactoryMock
+        });
 
         assert.deepEqual(result, {
             nodes: [{ name: '~pr' }, { name: '~commit' }, { name: 'foo' }, { name: 'bar' }],
             edges: [{ src: 'foo', dest: 'bar' }]
         });
     });
 
-    it('should handle logical OR requires', async () => {
-        const result = await getWorkflow(
-            {
-                jobs: {
-                    foo: { requires: ['~commit'] },
-                    A: { requires: ['foo'] },
-                    B: { requires: ['foo'] },
-                    C: { requires: ['~A', '~B', '~sd@1234:foo'] }
+    it('should handle stage requires', async () => {
```

Comment:
This test is focused on `AND` use-cases.

Per https://github.com/screwdriver-cd/data-schema/pull/510#discussion_r1260398901, for initial release, we will be supporting only `OR`. Can we have similar test to cover `OR` use-cases.